src/libraries/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/ldap/LdapConnection.cs

```diff
@@ -634,30 +634,39 @@ private int SendRequestHelper(DirectoryRequest request, ref int messageID)
                     }
 
                     // Process the attribute.
-                    string stringValue = null;
-                    if (assertion[0] is byte[] byteArray)
+                    byte[] byteArray;
+                    if (assertion[0] is string str)
                     {
-                        if (byteArray != null && byteArray.Length != 0)
-                        {
-                            berValuePtr = new BerVal
-                            {
-                                bv_len = byteArray.Length,
-                                bv_val = Marshal.AllocHGlobal(byteArray.Length)
-                            };
-                            Marshal.Copy(byteArray, 0, berValuePtr.bv_val, byteArray.Length);
-                        }
+                        var encoder = new UTF8Encoding();
+                        byteArray = encoder.GetBytes(str);
+                    }
+                    else if (assertion[0] is Uri uri)
+                    {
+                        var encoder = new UTF8Encoding();
+                        byteArray = encoder.GetBytes(uri.ToString());
+                    }
+                    else if (assertion[0] is byte[] bytes)
+                    {
+                        byteArray = bytes;
                     }
                     else
                     {
-                        stringValue = assertion[0].ToString();
+                        throw new ArgumentException(SR.ValidValueType);
```

Comment:
Can we also add a tests that ensures that this Argument Exception gets thrown?

Reply:
I had a look at this. Turns out DirectoryAttribute rejects other value types than string, Uri and byte[]. So this condition never happens.